clustered/server/src/main/java/org/ehcache/clustered/server/ServerSideServerStore.java

```diff
@@ -29,6 +29,6 @@
 public interface ServerSideServerStore extends ServerStore, MapInternals {
   void setEvictionListener(ServerStoreEvictionListener listener);
   ServerStoreConfiguration getStoreConfiguration();
-  List<OffHeapChainMap<Long>> getSegments();
+  List<? extends ChainMap<Long>> getSegments();
```

Comment:
So that's the method making the concrete implementation leak. Can't we instead really on a specific type for here?
From the usage, it seems we only need to get at a given `Set<K>` so a `List<Set<K>>` would be good enough.